src/credentials/examples/DeviceAttestationVerifierExample.cpp

```diff
@@ -0,0 +1,194 @@
+{
+public:
+    AttestationVerificationResult
+    VerifyAttestationInformation(const ByteSpan & attestation_info_buffer, const ByteSpan & attestation_challenge_buffer,
+                                 const ByteSpan & attestation_signature_buffer, const ByteSpan & pai_cert_der_buffer,
+                                 const ByteSpan & dac_cert_der_buffer, const ByteSpan & attestation_nonce) override;
+};
+
+AttestationVerificationResult ExampleDACVerifier::VerifyAttestationInformation(const ByteSpan & attestation_info_buffer,
+                                                                               const ByteSpan & attestation_challenge_buffer,
+                                                                               const ByteSpan & attestation_signature_buffer,
+                                                                               const ByteSpan & pai_cert_der_buffer,
+                                                                               const ByteSpan & dac_cert_der_buffer,
+                                                                               const ByteSpan & attestation_nonce)
+{
+    // match DAC and PAI VIDs
+    if (!pai_cert_der_buffer.empty())
+    {
+        VendorId paiVid;
+        VendorId dacVid;
+
+        VerifyOrReturnError(ExtractVIDFromX509Cert(pai_cert_der_buffer, paiVid) == CHIP_NO_ERROR,
+                            AttestationVerificationResult::kPaiFormatInvalid);
+        VerifyOrReturnError(ExtractVIDFromX509Cert(dac_cert_der_buffer, dacVid) == CHIP_NO_ERROR,
+                            AttestationVerificationResult::kDacFormatInvalid);
+
+        VerifyOrReturnError(paiVid == dacVid, AttestationVerificationResult::kDacVendorIdMismatch);
+    }
+
+    uint8_t akidBuf[Credentials::kKeyIdentifierLength];
+    MutableByteSpan akid(akidBuf);
+    ExtractAKIDFromX509Cert(pai_cert_der_buffer.empty() ? dac_cert_der_buffer : pai_cert_der_buffer, akid);
+
+    constexpr size_t paaCertAllocatedLen = kMaxDERCertLength;
+    chip::Platform::ScopedMemoryBuffer<uint8_t> paaCert;
+
+    VerifyOrReturnError(paaCert.Alloc(paaCertAllocatedLen), AttestationVerificationResult::kNoMemory);
+
+    MutableByteSpan paa(paaCert.Get(), paaCertAllocatedLen);
+    VerifyOrReturnError(GetProductAttestationAuthorityCert(akid, paa) == CHIP_NO_ERROR,
+                        AttestationVerificationResult::kPaaNotFound);
+
+    VerifyOrReturnError(ValidateCertificateChain(paa.data(), paa.size(), pai_cert_der_buffer.data(), pai_cert_der_buffer.size(),
+                                                 dac_cert_der_buffer.data(), dac_cert_der_buffer.size()) == CHIP_NO_ERROR,
+                        AttestationVerificationResult::kPaaUntrusted);
+
+    P256PublicKey remoteManufacturerPubkey;
+    VerifyOrReturnError(ExtractPubkeyFromX509Cert(dac_cert_der_buffer, remoteManufacturerPubkey) == CHIP_NO_ERROR,
+                        AttestationVerificationResult::kDacFormatInvalid);
```

Comment:
This check can come before full chain validation. If the signature against the DAC's associated private key does not succeed, then ValidateCertificateChain would have failed also. However, doing this check first helps with making finer-grained error checking.

Reply:
will update